src/test/java/core/basesyntax/RegistrationServiceTest.java

```diff
@@ -0,0 +1,149 @@
+        user = new User();
+        user.setLogin("Login1223");
+        user.setPassword("1234567");
+        user.setAge(20);
+        Assertions.assertEquals(user, service.register(user));
+        expectedSize += 2;
+        Assertions.assertEquals(expectedSize, Storage.people.size());
+    }
+
+    @Test
+    void addUserWithLoginLessSix_NotOk() {
+        user.setLogin("Login");
+        Assertions.assertThrows(RegistrationException.class, () -> service.register(user));
+        user = new User();
+        user.setLogin("Log");
+        user.setPassword("1234567");
+        user.setAge(20);
+        Assertions.assertThrows(RegistrationException.class, () -> service.register(user));
+    }
+
+    @Test
+    void addExistingUser_NotOk() {
+        Assertions.assertThrows(RegistrationException.class, () -> service.register(defaultUser));
+        User existingUser = new User();
+        existingUser.setId(defaultUser.getId());
+        existingUser.setLogin(defaultUser.getLogin());
+        existingUser.setPassword(defaultUser.getPassword());
+        existingUser.setAge(defaultUser.getAge());
+        Assertions.assertThrows(RegistrationException.class, () -> service.register(existingUser));
+    }
+
+    @Test
+    void addUserWithPasswordLessSix_NotOk() {
+        user.setPassword("12345");
+        Assertions.assertThrows(RegistrationException.class, () -> service.register(user));
+        user = new User();
+        user.setLogin("CorrectLogin");
+        user.setPassword("abc");
+        user.setAge(20);
+        Assertions.assertThrows(RegistrationException.class, () -> service.register(user));
+    }
+
+    @Test
+    void addUserWithPasswordEdgeLength_Ok() {
+        user.setPassword("123456");
+        Assertions.assertEquals(user, service.register(user));
+        expectedSize++;
+        Assertions.assertEquals(expectedSize, Storage.people.size());
+    }
+
+    @Test
+    void addUserWithNullValues_NotOk() {
+        User nullLoginUser = new User();
+        nullLoginUser.setPassword("123456");
+        nullLoginUser.setAge(20);
+        User nullPasswordUser = new User();
+        nullPasswordUser.setLogin("123456");
+        nullPasswordUser.setAge(20);
+        User nullAgeUser = new User();
+        nullAgeUser.setLogin("123456");
+        nullAgeUser.setPassword("password");
+        User nullValuesUser = new User();
+        Assertions.assertThrows(RegistrationException.class, () -> service.register(null));
+        Assertions.assertThrows(RegistrationException.class,
+                () -> service.register(nullValuesUser));
+        Assertions.assertThrows(RegistrationException.class,
+                () -> service.register(nullLoginUser));
+        Assertions.assertThrows(RegistrationException.class,
+                () -> service.register(nullPasswordUser));
+        Assertions.assertThrows(RegistrationException.class, () -> service.register(nullAgeUser));
+    }
```

Comment:
And this test you can divide on few separate notNull checks like: register_passwordNull_notOk, register_loginNull_notOk, register_ageNull_notOk